packages/ckeditor5-table/src/tablecolumnresize/tablecolumnresizeediting.js

```diff
@@ -643,28 +651,43 @@ export default class TableColumnResizeEditing extends Plugin {
 		const modelTable = modelLeftCell.findAncestor( 'table' );
 
 		const leftColumnIndex = getColumnEdgesIndexes( modelLeftCell, this._tableUtilsPlugin ).rightEdge;
-		const lastColumnIndex = this._tableUtilsPlugin.getColumns( modelTable ) - 1;
+		let lastColumnIndex = this._tableUtilsPlugin.getColumns( modelTable ) - 1;
 
-		const isRightEdge = leftColumnIndex === lastColumnIndex;
 		const isTableCentered = !modelTable.hasAttribute( 'tableAlignment' );
 		const isLtrContent = editor.locale.contentLanguageDirection !== 'rtl';
 
+		// We're enforcing the change of the resizers position using CSS, so the lastColumnIndex
+		// needs to be updated too. See #11785.
+		if ( ( modelTable.getAttribute( 'tableAlignment' ) === 'right' && isLtrContent ) ||
+			( modelTable.getAttribute( 'tableAlignment' ) === 'left' && !isLtrContent ) ) {
+			lastColumnIndex = 0;
+		}
+
+		const isDraggableEdge = leftColumnIndex === lastColumnIndex;
+
 		const viewTable = viewLeftCell.findAncestor( 'table' );
 		const viewFigure = viewTable.findAncestor( 'figure' );
 		const viewColgroup = [ ...viewTable.getChildren() ].find( viewCol => viewCol.is( 'element', 'colgroup' ) );
 		const viewLeftColumn = viewColgroup.getChild( leftColumnIndex );
-		const viewRightColumn = isRightEdge ? undefined : viewColgroup.getChild( leftColumnIndex + 1 );
+		let viewRightColumn = isDraggableEdge ? undefined : viewColgroup.getChild( leftColumnIndex + 1 );
 
 		const viewFigureParentWidth = getElementWidthInPixels( editor.editing.view.domConverter.mapViewToDom( viewFigure.parent ) );
 		const viewFigureWidth = getElementWidthInPixels( editor.editing.view.domConverter.mapViewToDom( viewFigure ) );
 		const tableWidth = getTableWidthInPixels( modelTable, editor );
 		const leftColumnWidth = columnWidths[ leftColumnIndex ];
-		const rightColumnWidth = isRightEdge ? undefined : columnWidths[ leftColumnIndex + 1 ];
+		let rightColumnWidth = isDraggableEdge ? undefined : columnWidths[ leftColumnIndex + 1 ];
+
+		// The data needs to be updated on right/left table alignment. See #11785.
+		if ( ( modelTable.getAttribute( 'tableAlignment' ) === 'right' && isLtrContent ) ||
+			( modelTable.getAttribute( 'tableAlignment' ) === 'left' && !isLtrContent ) ) {
+			viewRightColumn = isDraggableEdge ? undefined : viewColgroup.getChild( leftColumnIndex - 1 );
```

Comment:
As a side note the initial assignment:

```
let viewRightColumn = isTableResizeEdge ? undefined : viewColgroup.getChild( leftColumnIndex + 1 );
```

and reassignment in the `if`:

```
viewRightColumn = isTableResizeEdge ? undefined : viewColgroup.getChild( leftColumnIndex - 1 );
```

Differs only by using `+1` and `-1`.

So I suppose a better idea would be to just assign the offset to a constant and use it for both assignments.

```
cosnt nextColumnOffset = <condition> ? 1 : -1;
```

Reply:
@mateuszzagorski I have no answer to the question

> Why do we need to pick different right column in case of a table that is side aligned away?

I don't see rationale why are we setting `viewRightColumn` to `col` element describing column marked on the screen below:

![](https://user-images.githubusercontent.com/5353898/189688853-d72c2725-a65b-4265-b361-63a592167708.png)

I don't understand that.

Just thinking of it the thing that we should change here is [the `isTableResizeEdge` constant](https://github.com/ckeditor/ckeditor5/blob/aa0d99bd17c066d73518927157d7c4c7cfeda4be/packages/ckeditor5-table/src/tablecolumnresize/tablecolumnresizeediting.js#L668) - and I believe it should work already.

I have a feeling that we'll simply have to assume that "left" column might not always be availably, similar like https://github.com/ckeditor/ckeditor5/blob/aa0d99bd17c066d73518927157d7c4c7cfeda4be/packages/ckeditor5-table/src/tablecolumnresize/tablecolumnresizeediting.js#L680